File: client-side/client-side/src/services/userService.js
import axiosInstance from "../api/axiosInstance";

const userService = {
  // Get user by ID
  getUserById: async (id) => {
    const response = await axiosInstance.get(`/user/getUserById`, {
      params: { id },
    });
    console.log("axios instance ", axiosInstance)
    return response.data;
  },

  getUserWithPackages: async (id) => {
    const response = await axiosInstance.get(`/user/getUserWithPackages`, {
      params: { id },
    });
    console.log("axios instance ", axiosInstance)
    return response.data;
  },

  getUserByEmail: async () => {
    const response = await axiosInstance.get(`/user/getUserByEmail`);
    console.log("axios instance ", )
    return response.data;
  },

  // Get paginated users
  getPaginatedUsers: async (page = 0, size = 20) => {
    console.log("inside paginaed users api request ")
    const response = await axiosInstance.get(`/user/getPaginatedUsers`, {
      params: { page, size },
    });
    console.log("axios instance ")
    return response.data;
  },

  // Verify user with token
  verifyUser: async (token) => {
    const response = await axiosInstance.get(`/user/verify`, {
      params: { token },
    });
    return response.data;
  },

  // Resend verification email
  resendVerificationEmail: async (email) => {
    const response = await axiosInstance.post(`/user/resend-verification`, null, {
      params: { email },
    });
    return response.data;
  },
};

export default userService;

File: server-side/lms/src/main/java/com/totfd/lms/service/UserService.java
package com.totfd.lms.service;


import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
import com.totfd.lms.dto.user.request.UserRegisterDTO;
import com.totfd.lms.dto.auth.request.LoginRequestDTO;
import com.totfd.lms.dto.user.request.UserRequestDTO;
import com.totfd.lms.dto.auth.response.LoginResponseDTO;
import com.totfd.lms.dto.user.response.UserResponseDTO;
import com.totfd.lms.entity.Users;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface UserService {

    UserResponseDTO createUser(UserRequestDTO userRequestDTO);

    UserResponseDTO registerUser(UserRegisterDTO userRegisterDTO);

    LoginResponseDTO loginUser(LoginRequestDTO loginRequestDTO);

//    UserResponseDTO loginWithGoogle(GoogleIdToken.Payload payload);
    LoginResponseDTO loginWithGoogle(GoogleIdToken.Payload payload);

    UserResponseDTO getUserById(Long id);

    List<UserResponseDTO> getAllUsers();

    Page<UserResponseDTO> getAllUsersWithPage(Pageable pageable);

    UserResponseDTO updateUser(Long id, UserRequestDTO userRequestDTO);

    void deleteUser(Long id);

    String verifyUser(String Token);

    String resendVerificationEmail(String email);

    UserResponseDTO getByEmail(String email);

    Optional<Users> findByEmail(String email);
}
